src/test/java/io/appium/java_client/pagefactory_tests/widget/tests/combined/CombinedWidgetTest.java

```diff
@@ -34,29 +32,29 @@ public class CombinedWidgetTest extends WidgetTest {
      *
      * @return test parameters
      */
-    @Parameterized.Parameters
-    public static Collection<Object[]> data() {
-        return asList(
-                dataArray(new AppWithCombinedWidgets(),
-                    new AbstractStubWebDriver.StubAndroidDriver(), DefaultAndroidWidget.class),
-                dataArray(new AppWithCombinedWidgets(),
-                    new AbstractStubWebDriver.StubIOSXCUITDriver(), DefaultIosXCUITWidget.class),
-                dataArray(new AppWithCombinedWidgets(),
-                    new AbstractStubWebDriver.StubWindowsDriver(), DefaultWindowsWidget.class),
-                dataArray(new AppWithCombinedWidgets(),
-                    new AbstractStubWebDriver.StubBrowserDriver(), DefaultFindByWidget.class),
-                dataArray(new AppWithCombinedWidgets(),
-                    new AbstractStubWebDriver.StubAndroidBrowserOrWebViewDriver(), DefaultFindByWidget.class),
-                dataArray(new AppWithPartiallyCombinedWidgets(),
-                    new AbstractStubWebDriver.StubAndroidDriver(), DefaultAndroidWidget.class),
-                dataArray(new AppWithPartiallyCombinedWidgets(),
-                    new AbstractStubWebDriver.StubIOSXCUITDriver(), DefaultStubWidget.class),
-                dataArray(new AppWithPartiallyCombinedWidgets(),
-                    new AbstractStubWebDriver.StubWindowsDriver(), DefaultStubWidget.class),
-                dataArray(new AppWithPartiallyCombinedWidgets(),
-                    new AbstractStubWebDriver.StubBrowserDriver(), DefaultFindByWidget.class),
-                dataArray(new AppWithPartiallyCombinedWidgets(),
-                    new AbstractStubWebDriver.StubAndroidBrowserOrWebViewDriver(), DefaultFindByWidget.class)
+
+    public static Stream<Arguments> data() {
```

Comment:
shouldn't this method be referenced in tests via `@MethodSource` annotation

Reply:
Yes, I have added `@MethodSource` annotation in `WidgetTest` which is parent class for `CombinedWidgetTest` and  `CombinedAppTest` as follows
```java
    @ParameterizedTest
    @MethodSource("data")
    public abstract void checkThatWidgetsAreCreatedCorrectly();
```